pandas/core/internals/managers.py

```diff
@@ -1409,29 +1411,37 @@ def take(self, indexer, axis: int = 1, verify: bool = True, convert: bool = True
             new_axis=new_labels, indexer=indexer, axis=axis, allow_dups=True
         )
 
-    def equals(self, other) -> bool:
+    def equals(self, other: "BlockManager") -> bool:
         self_axes, other_axes = self.axes, other.axes
         if len(self_axes) != len(other_axes):
             return False
         if not all(ax1.equals(ax2) for ax1, ax2 in zip(self_axes, other_axes)):
             return False
-        self._consolidate_inplace()
-        other._consolidate_inplace()
-        if len(self.blocks) != len(other.blocks):
-            return False
 
-        # canonicalize block order, using a tuple combining the mgr_locs
-        # then type name because there might be unconsolidated
-        # blocks (say, Categorical) which can only be distinguished by
-        # the iteration order
-        def canonicalize(block):
-            return (block.mgr_locs.as_array.tolist(), block.dtype.name)
-
-        self_blocks = sorted(self.blocks, key=canonicalize)
-        other_blocks = sorted(other.blocks, key=canonicalize)
-        return all(
-            block.equals(oblock) for block, oblock in zip(self_blocks, other_blocks)
-        )
+        if self.ndim == 1:
+            if other.ndim != 1:
+                return False
+            blk = self.blocks[0]
+            rblk = other.blocks[0]
+            if blk.is_extension and rblk.is_extension:
+                return blk.values.equals(rblk.values)
+            elif blk.is_extension or rblk.is_extension:
+                return False
+            else:
+                return array_equivalent(blk.values, rblk.values)
+
+        for i in range(len(self.items)):
+            left = self.iget_values(i)
+            right = other.iget_values(i)
+            if not is_dtype_equal(left.dtype, right.dtype):
+                return False
+            elif isinstance(left, ExtensionArray):
+                if not left.equals(right):
+                    return False
```

Comment:
this can be `return left.equals(right)` ?

And same for `array_equivalent` below?

Reply:
yes, will update

Reply:
no, wait, not here.  this is an "all" check, so we only return on False.

Reply:
Ah, yes, it's inside the for loop ..